### DISCUSSION TITLE
custom getPresignedObjectUrl method returned value's host

### DISCUSSION THREAD

**Discussion by Archsx** [minio/minio#16190] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Say we only have one server, nginx listened at port `80`,

 frontend need to call some api like this to get the upload image url,
```
http://xx.xx.xx.xx:80/api/getpresignedurl/fileName
```
the api service located at port `8080` and the request is transform from port `80` to it by nginx, and our minio service start at port `9000`,  looked at the code bellow:
```
            val url =
                mc.getPresignedObjectUrl(
                    GetPresignedObjectUrlArgs.builder()
                        .method(Method.PUT)
                        .bucket(bucketName)
                        .`object`(fileName)
                        .expiry(24 * 60 * 60)
                        .build()
                )

```
this is the url we return to frontend, however, it is always like this:
```
http://xx.xx.xx.xx:9000/bucketName/foo1.jpg?X-Amz-Algorithm=......
```
when frontend use this url to upload image, they would get *Cross Origin* error,  I tried to add some response header via nginx to avoid cross origin error 

but wait, the nginx listen at 80, the url is point to port 9000, it seems to be impossible to do that. 

in a nut shell, why the method `getPresignedObjectUrl` always return a url whose host part is the one when we tried to initialize minioclient,

```
%prod.quarkus.minio.url=https://xxx.xxx.xxx.xxx:9000
```

**Describe the solution you'd like**
if the method `GetPresignedObjectUrlArgs` have a mothod like `host`, it would be great:
```
                    GetPresignedObjectUrlArgs.builder()
                        .host("xx.xx.xx.xx:80") // it would be great
                        .method(Method.PUT)
                        .bucket(bucketName)
                        .`object`(fileName)
                        .expiry(24 * 60 * 60)
                        .build()

```

**Additional context**
Anyway, I just wanna a clear cross origin solution .

**Comment by harshavardhana** [comment, marked as answer]
You need to initialize the client constructor with your Public IP (your load balancer) not the internal IP on port 9000

**Reply by Archsx** [reply]
could you please tell me more about it ? say we our server at 123.123.123.123, and we only have *one* server, then the configuration should be like:
```
// the nginx is listen at 80
%prod.quarkus.minio.url=http://123.123.123.123:80
```
but  then I would get error when trying to initialize the minioclient, and it make sense , cause the minio service  runs at port `9000`

**Comment by lunxian8** [comment]
What if the api service can't access the nginx load balancer? A computer A deploies services, and a computer B performs gateway processing. If b can access the public network, the domain name associated with B configured by A cannot be instantiated
